### PR TITLE
Add release type to CI matrix

### PR DESCRIPTION
**Pull request by luqasz**
Catch possible errors early in CI with release and debug builds.

### CODE REVIEW

**Comment by gnif** [comment]
Looks good, but please use the following commit message format:
`[section] part: description`
ie:
`[all] github: add release type to CI matrix`

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -25,7 +28,8 @@ jobs:
       run: |
         mkdir client/build
         cd client/build
-        CC=/usr/bin/${{ matrix.cc }} cmake ..
+        CC=/usr/bin/${{ matrix.cc }}
+        cmake -DCMAKE_RELEASE_TYPE={{ matrix.release_type }} ..
```
Comment (quantum5) [inline .github/workflows/build.yml:32]: It's actually `CMAKE_BUILD_TYPE`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -25,7 +28,8 @@ jobs:
       run: |
         mkdir client/build
         cd client/build
-        CC=/usr/bin/${{ matrix.cc }} cmake ..
+        CC=/usr/bin/${{ matrix.cc }}
```
Comment (quantum5) [inline .github/workflows/build.yml:31]: You can't put this on a separate line unless you prefix it with `export`. The CI build is currently using `/usr/bin/cc` as the C compiler for both GCC and Clang builds.

**Review by quantum5 — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (luqasz) [reply .github/workflows/build.yml]: Fixed.

File: .github/workflows/build.yml (hunk 2 above)
Comment (luqasz) [reply .github/workflows/build.yml]: Fixed.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -6,6 +6,9 @@ jobs:
     strategy:
       matrix:
         cc: [gcc, clang]
+        release_type:
```
Comment (quantum5) [inline .github/workflows/build.yml:9]: Maybe rename this? Otherwise looks fine.

Comment (luqasz) [reply .github/workflows/build.yml:9]: Isn't it already named `release_type` ? IMO self explanatory.

Comment (quantum5) [reply .github/workflows/build.yml:9]: It makes more sense to be called `build_type`.

Comment (gnif) [reply .github/workflows/build.yml:9]: Yes, `build_type` for consistency please.

Comment (luqasz) [reply .github/workflows/build.yml]: done

**Pull request merged**